modules/core/src/effects/mask/utils.js

```diff
@@ -0,0 +1,75 @@
+import {Matrix4} from '@math.gl/core';
+import {pixelsToWorld} from '@math.gl/web-mercator';
+import {WebMercatorViewport} from '@deck.gl/core';
+
+/*
+ * Compute orthographic projection that converts shader common space coordinates into mask clipspace
+ */
+export function getMaskProjectionMatrix({width, height, pixelUnprojectionMatrix}) {
+  // Unproject corners of viewport into common space (these correspond to the edges
+  // of the framebuffer we are rendering into)
+  const [[left, top], [right, bottom]] = [
+    [0, 0],
+    [width, height]
+  ].map(pixel => pixelsToWorld(pixel, pixelUnprojectionMatrix));
+
+  // Construct orthographic projection that will map common space positions into clipspace
+  const near = 0.1;
+  const far = 1000;
+  return new Matrix4().ortho({left, top, right, bottom, near, far});
+}
+
+/*
+ * Compute viewport to render the mask into, such that it covers an
+ * area currently visible (extended by a buffer) or the area of the masking
+ * data, whichever is smaller
+ */
+export function getMaskViewport(positions, layerViewport, {width, height}) {
+  const bounds = layerViewport.getBounds();
+  let viewport = new WebMercatorViewport({width, height}).fitBounds([
```

Comment:
Is there a reason to hard code the viewport type here?

Reply:
The other viewports don't have the `fitBounds` methods and the rest of the code doesn't support other projections. Would it be acceptable to restrict this PR to the `WebMercatorViewport` as it the mainline case and add the other viewports in a followup PR?

Reply:
I've fixed this now, such that `WebMercatorViewport` isn't hardcoded. If `fitBounds` is available on the `Viewport`, then it is used for the calculation, otherwise a fallback method is used